cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaAgentClient.java

```diff
@@ -0,0 +1,145 @@
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.operator.cluster.model.DnsNameGenerator;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.Util;
+
+/**
+ * Creates HTTP client and interacts with Kafka Agent's REST endpoint
+ */
+class KafkaAgentClient {
+
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(KafkaAgentClient.class.getName());
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private static final String BROKER_STATE_REST_PATH = "/v1/broker-state/";
+    private static final int BROKER_STATE_HTTPS_PORT = 8443;
+
+    private String namespace;
+    private Reconciliation reconciliation;
+    private String cluster;
+    private Secret clusterCaCertSecret;
+    private Secret coKeySecret;
+    private HttpClient httpClient;
+
+    KafkaAgentClient(Reconciliation reconciliation, String cluster, String namespace, Secret clusterCaCertSecret, Secret coKeySecret) {
+        this.reconciliation = reconciliation;
+        this.cluster = cluster;
+        this.namespace = namespace;
+        this.clusterCaCertSecret = clusterCaCertSecret;
+        this.coKeySecret = coKeySecret;
+        this.httpClient = createHttpClient();
+    }
+
+    /* test */ KafkaAgentClient(Reconciliation reconciliation, String cluster, String namespace) {
+        this.reconciliation = reconciliation;
+        this.namespace = namespace;
+        this.cluster =  cluster;
+    }
+
+    private HttpClient createHttpClient() {
+        if (clusterCaCertSecret == null || coKeySecret == null) {
+            throw new RuntimeException("Missing secrets for cluster CA and operator certificates required to create connection to Kafka Agent");
+        }
+
+        try {
+            SSLContext sslContext = SSLContext.getInstance("TLS");
+            KeyStore trustStore = KeyStore.getInstance("PKCS12");
+            trustStore.load(new ByteArrayInputStream(
+                            Util.decodeFromSecret(clusterCaCertSecret, "ca.p12")),
+                    new String(Util.decodeFromSecret(clusterCaCertSecret, "ca.password"), StandardCharsets.UTF_8).toCharArray());
+
+            String trustManagerFactoryAlgorithm = TrustManagerFactory.getDefaultAlgorithm();
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(trustManagerFactoryAlgorithm);
+            trustManagerFactory.init(trustStore);
+            char[] keyPassword = new String(Util.decodeFromSecret(coKeySecret, "cluster-operator.password"), StandardCharsets.UTF_8).toCharArray();
+            KeyStore coKeyStore = KeyStore.getInstance("PKCS12");
+            coKeyStore.load(new ByteArrayInputStream(
+                            Util.decodeFromSecret(coKeySecret, "cluster-operator.p12")),
+                    keyPassword
```

Comment:
why you used a variable here instead of inlining the password recovery as for the truststore?

Reply:
It's also used in line 89 for initialising the keyManagerFactory.

Reply:
Oh thanks, I missed that!